gusto/forcing.py

```diff
@@ -221,212 +220,91 @@ def _build_forcing_solver(self):
+        L -= self.scaling*dbdy*eady_exp*inner(self.test,as_vector([0.,1.,0.]))*dx
+        return L
 
-        bcs = [DirichletBC(Vu, 0.0, "bottom"),
-               DirichletBC(Vu, 0.0, "top")]
+    def _build_forcing_solvers(self):
 
-        u_forcing_problem = LinearVariationalProblem(
-            a,L,self.uF, bcs=bcs
-        )
-
-        self.u_forcing_solver = LinearVariationalSolver(u_forcing_problem)
+        super(EadyForcing, self)._build_forcing_solvers()
 
         # b_forcing
-
-        Vb = state.spaces("HDiv_v")
-
+        dbdy = self.state.parameters.dbdy
+        Vb = self.state.spaces("HDiv_v")
         F = TrialFunction(Vb)
         gamma = TestFunction(Vb)
         self.bF = Function(Vb)
+        u0, _, b0 = split(self.x0)
 
         a = gamma*F*dx
-        L = -gamma*self.scaling*(dbdy*inner(u0,as_vector([0.,1.,0.])))*dx
+        L = -self.scaling*gamma*(dbdy*inner(u0, as_vector([0.,1.,0.])))*dx
 
         b_forcing_problem = LinearVariationalProblem(
-            a,L,self.bF
+            a, L, self.bF
         )
 
         self.b_forcing_solver = LinearVariationalSolver(b_forcing_problem)
 
-        # divergence_free
-
-        Vp = state.spaces("DG")
-        p = TrialFunction(Vp)
-        q = TestFunction(Vp)
-        self.divu = Function(Vp)
-
-        a = p*q*dx
-        L = q*div(u0)*dx
-
-        divergence_problem = LinearVariationalProblem(
-            a, L, self.divu)
-
-        self.divergence_solver = LinearVariationalSolver(divergence_problem)
-
     def apply(self, scaling, x_in, x_nl, x_out, **kwargs):
 
-        self.x0.assign(x_nl)
-        self.scaling.assign(scaling)
-        if 'mu_alpha' in kwargs and kwargs['mu_alpha'] is not None:
-            self.mu_scaling.assign(kwargs['mu_alpha'])
-        self.u_forcing_solver.solve()  # places forcing in self.uF
+        super(EadyForcing, self).apply(scaling, x_in, x_nl, x_out, **kwargs)
         self.b_forcing_solver.solve()  # places forcing in self.bF
-
-        u_out, p_out, b_out = x_out.split()
-
-        x_out.assign(x_in)
-        u_out += self.uF
+        _, _, b_out = self.x_out.split()
         b_out += self.bF
 
-        if kwargs.get("incompressible", False):
-            self.divergence_solver.solve()
-            p_out.assign(self.divu)
-
 
-class CompressibleEadyForcing(Forcing):
+class CompressibleEadyForcing(CompressibleForcing):
     """
     Forcing class for compressible Eady equations.
     """
 
-    def _build_forcing_solver(self):
-        """
-        Only put forcing terms into the u equation.
-        """
-
-        state = self.state
-        self.scaling = Constant(1.)
-        Vu = state.spaces("HDiv")
-        W = state.W
-
-        self.x0 = Function(W)   # copy x to here
-
-        u0,rho0,theta0 = split(self.x0)
-
-        # u_forcing
+    def forcing_term(self):
 
-        F = TrialFunction(Vu)
-        w = TestFunction(Vu)
-        self.uF = Function(Vu)
+        # L = super(EadyForcing, self).forcing_term()
```

Comment:
I don't quite understand why this doesn't work - should I not expect super to find the forcing_term method because it is defined in the grandparent, rather than the parent, class? Is my solution of calling Forcing.forcing_term ok?